blas/impl/KokkosBlas2_ger_impl.hpp

```diff
@@ -0,0 +1,339 @@
+  KOKKOS_INLINE_FUNCTION void operator()( TwoLevelGER_LayoutLeftTag
+                                        , const member_type & team
+                                        ) const {
+    using KAT = Kokkos::ArithTraits<typename AViewType::non_const_value_type>;
+
+    if (alpha_ == KAT::zero()) {
+      // Nothing to do
+    }
+    else {
+      const IndexType M ( A_.extent(0) );
+      const IndexType j ( team.league_rank() );
+      if (justTranspose_) {
+        const A_value_type y_fixed( y_(j) );
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, M), [&](const IndexType & i) {
+          A_(i,j) += A_value_type( alpha_ * x_(i) * y_fixed );
+        });
+      }
+      else {
+        const A_value_type y_fixed( KAT::conj( y_(j) ) );
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, M), [&](const IndexType & i) {
+          A_(i,j) += A_value_type( alpha_ * x_(i) * y_fixed );
+        });
+      }
+    }
+  }
+
+  // LayoutRight version: one team per row
+  KOKKOS_INLINE_FUNCTION void operator()( TwoLevelGER_LayoutRightTag
+                                        , const member_type & team
+                                        ) const {
+    using KAT = Kokkos::ArithTraits<typename AViewType::non_const_value_type>;
+
+    if (alpha_ == KAT::zero()) {
+      // Nothing to do
+    }
+    else {
+      const IndexType    N      ( A_.extent(1) );
+      const IndexType    i      ( team.league_rank() );
+      const A_value_type x_fixed( x_(i) );
+      if (justTranspose_) {
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, N), [&](const IndexType & j) {
+          A_(i,j) += A_value_type( alpha_ * x_fixed * y_(j) );
+        });
+      }
+      else {
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, N), [&](const IndexType & j) {
+          A_(i,j) += A_value_type( alpha_ * x_fixed * KAT::conj( y_(j) ) );
+        });
+      }
+    }
+    team.team_barrier();
+  }
+
+private:
+  bool                           justTranspose_;
+  AlphaCoeffType                 alpha_;
+  typename XViewType::const_type x_;
+  typename YViewType::const_type y_;
+  AViewType                      A_;
+};
+
+// Two-level parallel version of GER.
+template <class XViewType, class YViewType, class AViewType,
+          class IndexType = typename AViewType::size_type>
+void twoLevelGer( const typename AViewType::execution_space  & space
+                , const          char                          trans[]
+                , const typename AViewType::const_value_type & alpha
+                , const          XViewType                   & x
+                , const          YViewType                   & y
+                , const          AViewType                   & A
+                ) {
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF( "Entering IMPL twoLevelGer(), AViewType = %s\n", typeid(AViewType).name() );
+  static_assert(Kokkos::is_view<XViewType>::value, "XViewType must be a Kokkos::View.");
```

Comment:
I think all these have already been checked in `KokkosBlas2_ger.hpp`, not sure why we check again here?

Reply:
Agreed. Now there is only one check at blas/src/KokkosBlas2_ger.hpp